### PR TITLE
Pre-abelian

### PR DESCRIPTION
**Pull request by eggrobin**
Lots of cleanup in the C#, bugfixes, version logging in the C++ (but not in the C#).

### CODE REVIEW

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 1]
```diff
@@ -33,6 +35,12 @@ public partial class PluginAdapter : UnityEngine.MonoBehaviour {
   private int first_selected_celestial_ = 0;
   private int second_selected_celestial_ = 0;
 
+  private bool time_is_advancing_;
+
+  private DateTime last_plugin_reset_;
+
+  private static bool an_instance_is_loaded;
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Trailing `_`.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 2]
```diff
@@ -70,155 +78,195 @@ public partial class PluginAdapter : UnityEngine.MonoBehaviour {
...
+
+  private void UpdateVessel(Vessel vessel, double universal_time) {
+    bool inserted = InsertOrKeepVessel(
+        plugin_,
+        vessel.id.ToString(),
+        vessel.orbit.referenceBody.flightGlobalsIndex);
+    if (inserted) {
+      SetVesselStateOffset(plugin               : plugin_,
+                           vessel_guid          : vessel.id.ToString(),
+                           from_parent : new QP {q = (XYZ)vessel.orbit.pos,
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: No space after `QP`, align colons.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 3]
```diff
@@ -70,155 +78,195 @@ public partial class PluginAdapter : UnityEngine.MonoBehaviour {
...
+        vessel.id.ToString(),
+        vessel.orbit.referenceBody.flightGlobalsIndex);
+    if (inserted) {
+      // NOTE(egg): this is only used when a (plugin-managed) physics bubble
+      // appears with a new vessel (e.g. when exiting the atmosphere).
+      // TODO(egg): these degrees of freedom are off by one Δt and we don't
+      // compensate for the pos/vel synchronization bug.
+      SetVesselStateOffset(plugin      : plugin_,
+                           vessel_guid : vessel.id.ToString(),
+                           from_parent : new QP {q = (XYZ)vessel.orbit.pos,
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: No space after `QP`.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 4]
```diff
@@ -322,16 +387,53 @@ public partial class PluginAdapter : UnityEngine.MonoBehaviour {
...
+          "Plugin was started at " +
+          last_plugin_reset_.ToUniversalTime().ToString("O");
     }
+    UnityEngine.GUILayout.TextArea(last_reset_information);
+    bool barycentric_rotating =
+        first_selected_celestial_ != second_selected_celestial_;
+    String reference_frame_description =
+        "The trajectory of the active vessel is plotted in ";
+    if (barycentric_rotating) {
+      reference_frame_description =
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: +=

File: ksp_plugin_adapter/ksp_plugin_adapter.cs [hunk 5]
```diff
@@ -322,16 +387,53 @@ public partial class PluginAdapter : UnityEngine.MonoBehaviour {
...
+          "the reference frame fixing the barycentre of " +
+          FlightGlobals.Bodies[first_selected_celestial_].theName + " and " +
+          FlightGlobals.Bodies[second_selected_celestial_].theName + ", " +
+          "the line through them, and the plane in which they move about the " +
+          "barycentre.";
+    } else {
+      reference_frame_description =
```
Comment (pleroy) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: +=

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 4 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

File: ksp_plugin_adapter/ksp_plugin_adapter.cs (hunk 5 above)
Comment (eggrobin) [inline ksp_plugin_adapter/ksp_plugin_adapter.cs]: Done.

**Pull request merged**